cmd/appliance/seed.go

```diff
@@ -0,0 +1,276 @@
+					"activated": "false",
+				},
+			}
+			switch len(args) {
+			// no arguments will provider a interactive experience for the user, if TTY support it
+			case 0:
+				if noInteractive || !opts.CanPrompt {
+					return errors.New("can't prompt, You need to provide all arguments")
+				}
+				applianceID, err := appliancepkg.PromptSelectAll(ctx, a, filter)
+				if err != nil {
+					return err
+				}
+				opts.applianceID = applianceID
+				format := []string{
+					jsonFormat: "JSON format",
+					isoFormat:  "ISO format",
+				}
+				qs := &survey.Select{
+					PageSize: len(format),
+					Message:  "Seed type:",
+					Options:  format,
+				}
+
+				selectedFormat := 0
+				if err := prompt.SurveyAskOne(qs, &selectedFormat, survey.WithValidator(survey.Required)); err != nil {
+					return err
+				}
+				opts.format = seedformat(selectedFormat)
+				if opts.format == isoFormat {
+					opts.iso = true
+				}
+				methods := []string{
+					cloud:      "Use SSH key provided by the cloud instance",
+					publicKey:  "Use SSH public key",
+					passphrase: "Use Password",
+				}
+				qs = &survey.Select{
+					PageSize: len(methods),
+					Message:  "SSH Authentication Method:",
+					Options:  methods,
+				}
+				selectedIndex := 0
+				if err := prompt.SurveyAskOne(qs, &selectedIndex, survey.WithValidator(survey.Required)); err != nil {
+					return err
+				}
+				switch selectedIndex {
+				case int(cloud):
+					opts.provideCloudSSHKey = true
+				case int(publicKey):
+					qs := &survey.Input{
+						Message: "path to public ssh key:",
+					}
+					if err := prompt.SurveyAskOne(qs, &opts.sshKey); err != nil {
+						return err
+					}
+				case int(passphrase):
+					hasStdin := false
+					stat, err := os.Stdin.Stat()
+					if err == nil && (stat.Mode()&os.ModeCharDevice) == 0 {
+						hasStdin = true
+					}
+					password, err := prompt.GetPassphrase(opts.In, opts.CanPrompt, hasStdin, "Appliance's CZ user password:")
+					if err != nil {
+						return err
+					}
+					opts.password = password
+				}
+
+			case 1:
+				if !util.IsUUID(args[0]) {
+					return errors.New("expected appliance UUID")
+				}
+				opts.applianceID = args[0]
+
+			}
+
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			return seedRun(c, args, &opts)
+		},
+	}
+	cmd.Flags().BoolVar(&opts.provideCloudSSHKey, "provide-cloud-ssh-key", false, "Tells appliance to use the key generated by AWS or Azure")
+	cmd.Flags().BoolVar(&opts.allowCustomization, "allow-customization", false, "Whether the Appliance should allow customizations or not")
+	cmd.Flags().BoolVar(&opts.iso, "iso-format", false, "Export as ISO format")
+	cmd.Flags().BoolVar(&opts.json, "json", false, "output as json to stdout, only applicable in json seed type")
+	cmd.Flags().StringVarP(&opts.sshKey, "ssh-key", "", "", "filepath to public ssh key to be allowed")
+	cmd.Flags().IntVarP(&opts.validityDays, "validity-days", "", 1, "How many days the seed should be valid for")
```

Comment:
This could be a time duration flag as well, which would provide a lot more flexibility. Of course that would be useless if the seed validity can only be n days.